packages/app-core/src/alerting/fifo.ts

```diff
@@ -0,0 +1,63 @@
+
+export class RingFifo<T> {
+	protected store: T[];
+	protected head = 0;
+	protected tail = 0;
+	protected full = false;
```

Comment:
I think it would be safer to have a method with the head and tail check rather than storing this variable. It would reduce the chance of a change being made that allows this to go out of sync, since it needs to be set in different places.

Reply:
Great suggestion! We just discussed and the reason I went with a property is because `head == tail` when the queue is empty and when the queue is full! Otherwise we could over-allocate by 1 element